tests/Doctrine/Tests/ORM/Query/ParserTest.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+namespace Doctrine\Tests\ORM\Query;
+
+use Doctrine\ORM\Query;
+use Doctrine\ORM\Query\Lexer;
+use Doctrine\ORM\Query\Parser;
+
+class ParserTest extends \Doctrine\Tests\OrmTestCase
+{
+
+    /**
+     * @covers \Doctrine\ORM\Query\Parser::AbstractSchemaName
```

Comment:
I wanted to even limit it to the method level to make sure I've got relevant parts covered.